tensorflow/python/keras/distribute/distributed_training_utils.py

```diff
@@ -769,7 +777,8 @@ def _upcast_low_precision_outputs(output):
         sample_weight_mode=model.sample_weight_mode,
         weighted_metrics=metrics_module.clone_metrics(
             model._compile_weighted_metrics),
-        target_tensors=targets)
+        target_tensors=targets,
+        **model._function_kwargs)
```

Comment:
This is using a private field of keras model, which is subject to change. Also we removed kwargs for compile for the latest model, so this is very likely to break.

Btw, what was expected in the kwargs?

Reply:
It may contain `'feed_dict', 'fetches', 'options', 'run_metadata'` as `kwargs` in the old `compile`.
And for this pr, the `options` and `run_metadata` is needed.

Reply:
I think my point here is that the model._ _function_kwargs may not exist in the latest version of model code. Relying on a Keras private field seems not ideal.

See latest model code in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/engine/training.py#L458

Reply:
I've just read the code for the new `compile` function. Could you tell me will keras still support `run_metadata` or `timeline` in the future release (or maybe after separated from tensorflow if that rfc is relevant to the problem we are discussing now)?

Reply:
I think we probably will remove them and kwargs in future, since this is usually source of error where unrecognized argument gets caught there. We previously had issue that people made typo in the argument name, and it just get ignored silently. We recently did similar change to remove the kwargs in model.fit(), and I would expect to take same actions for compile as well.

See https://github.com/tensorflow/tensorflow/commit/fb1b76c7e3a159fb02a0a8a62367798760bb6381

For run_metadata particularly, it is used in session and funcgraph, which I think is not actively used in TF v2 keras.